### PR TITLE
Add topologySpreadConstraints

### PR DESCRIPTION
**Pull request by jbhalodia-slack**
## Purpose of this PR
Its good to spread the Spark Operator pods across the cluster among failure-domains such as regions, zones, nodes, and other user-defined topology domains. This can help to achieve high availability as well as efficient resource utilization. 

**Proposed changes:**
- Set [topologySpreadConstraints](https://kubernetes.io/docs/concepts/scheduling-eviction/topology-spread-constraints/) within the charts

## Change Category
Indicate the type of change by marking the applicable boxes:

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] Feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that could affect existing functionality)
- [ ] Documentation update

### Rationale

Production workloads should use enable [topologySpreadConstraints](https://kubernetes.io/docs/concepts/scheduling-eviction/topology-spread-constraints/#motivation) to make sure their workloads are running in HA and are resilient to node or AZ specific failures.

## Checklist
Before submitting your PR, please review the following:

- [x] I have conducted a self-review of my own code.
- [x] I have updated documentation accordingly.
- [x] I have added tests that prove my changes are effective or that my feature works.
- [x] Existing unit tests pass locally with my changes.

### Additional Notes

Github Issue: https://github.com/kubeflow/spark-operator/issues/2086
Slack Thread: https://cloud-native.slack.com/archives/C074588U7EG/p1721240818494049

### CODE REVIEW

File: charts/spark-operator-chart/tests/poddisruptionbudget_test.yaml [hunk 1]
```diff
@@ -17,21 +17,22 @@ tests:
 
   - it: Should render spark operator podDisruptionBudget if podDisruptionBudget.enable is true
     set:
+      replicaCount: 2
```
Comment (jbhalodia-slack) [inline charts/spark-operator-chart/tests/poddisruptionbudget_test.yaml:20]: PDB tests were failing on the master branch so these are fixes to get them to pass.

**Comment by jbhalodia-slack** [comment]
Hi  @vara-bonthu @andreyvelich @ChenYi015 @yuchaoran2011, could you please review this PR? 🙇‍♂️

**Review by vara-bonthu — APPROVED** [review]
Thanks for the PR @jbhalodia-slack 
/approve

@yuchaoran2011 @ChenYi015 Please review

**Review by ChenYi015 — APPROVED** [review]
Thanks for the contribution! 
/lgtm

**Pull request merged**